core/src/raw/oio/write.rs

```diff
@@ -178,3 +180,87 @@ impl<T: BlockingWrite + ?Sized> BlockingWrite for Box<T> {
         (**self).close()
     }
 }
+
+/// AppendOperation is the name for APIs of Append.
+#[derive(Debug, Copy, Clone, Hash, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum AppendOperation {
```

Comment:
Maybe we could pick `append` into a new file.

Reply:
Seems from the implementation level, append is parallel with write. I think separate this into a new file makes more sense.

Reply:
LGTM.